java/java-jersey-jaxrs/src/main/java/io/swagger/sample/data/PetData.java

```diff
@@ -68,11 +69,13 @@ public Pet getPetById(long petId) {
 
```

Comment:
So you in almost all methods synchronize on `pets´, just not in `getPetById` (some lines up from here). Intentionally?
I'm not sure if this here could create some issue ... 

The iterator is not synchronized, but checks for concurrent modifications (in an unsynchronized way).
So if any other request gets in between with deletion/addition, this could throw an exception.

On the other hand, if you synchronize everything anyways, you could get rid of the synchronizedList.